boot/bootutil/include/bootutil/bootutil.h

```diff
@@ -76,7 +77,7 @@ struct image_trailer {
 };
 
 /* you must have pre-allocated all the entries within this structure */
-int boot_go(struct boot_rsp *rsp);
+int boot_go(int img_cnt, struct boot_rsp *rsp);
```

Comment:
This API change is at the heart of the new work here.

There is a design document in the doc/ directory. It is harder to review these changes without a corresponding patch to that document which declares the high-level assumptions being made.

This series should include a design doc update as well, to make sure that the code changes correspond to the high-level proposal.